src/DataStructures/Matrix.hpp

```diff
@@ -25,10 +28,18 @@ class Matrix {
  public:
   /// Create and set each element to value
   Matrix(size_t number_of_rows, size_t number_of_colums,
-         double value = std::numeric_limits<double>::signaling_NaN());
+         double value = std::numeric_limits<double>::signaling_NaN()) noexcept;
+
+  /// Create a matrix from a two-dimensional vector of rows x columns. Requires
+  /// that all rows have the same size.
+  explicit Matrix(const std::vector<std::vector<double>>& rows);
 
   /// Default constructor needed for serialization
-  Matrix() = default;
+  Matrix() noexcept = default;
+
+  // I put this here for now.
```

Comment:
clarification please (where is it intended to go?)

Reply:
Do you have a suggestion? We need functionality to apply a `Matrix` only to parts of a `DataVector` for situations where the `DataVector` is a tensor product between multiple dimensions and the `Matrix` should be applied in a particular dimension.

Reply:
I just wanted clarification on the comment, I thought you might have other ideas about where it should go or it was going to be moved in a future commit or something like that.

Reply:
I see. This is quite open for discussion I think :)